src/runtime/virtcontainers/clh.go

```diff
@@ -726,10 +726,16 @@ func (clh *cloudHypervisor) StartVM(ctx context.Context, timeout int) error {
 		return err
 	}
 	defer func() {
-		if err != nil {
-			if shutdownErr := clh.stopVirtiofsDaemon(ctx); shutdownErr != nil {
-				clh.Logger().WithError(shutdownErr).Warn("error shutting down VirtiofsDaemon")
-			}
+		if err == nil {
+			return
+		}
+
+		if clh.config.SharedFS == config.NoSharedFS {
```

Comment:
I prefer to use `q.config.SharedFS == config.VirtioFS || q.config.SharedFS == config.VirtioFSNydus`, which means that this method should be called if virtio-fs is used. With your current version, if using virtio-9p, `stopVirtiofsDaemon` will still be called unexpectedly, right?

```go
if q.config.SharedFS == config.VirtioFS || q.config.SharedFS == config.VirtioFSNydus {
		if shutdownErr := clh.stopVirtiofsDaemon(ctx); shutdownErr != nil {
			clh.Logger().WithError(shutdownErr).Warn("error shutting down VirtiofsDaemon")
		}
}